R/geotargets-option.R

```diff
@@ -0,0 +1,65 @@
+#' Get or Set geotargets Options
+#'
+#' Get or set behavior for geospatial data target stores using geotargets-specific global options.
+#'
+#' @param option_name Character. Option name. See Details.
+#'
+#' @details
+#'
+#' ## Available Options
+#'
+#'  - `"geotargets.gdal.raster.creation_options"` - set the GDAL creation options used when writing raster files to target store (default: `"ENCODING=UTF-8"`)
+#'
+#'  - `"geotargets.gdal.raster.driver_name"` - set the file type used for raster data in target store (default: `"GTiff"`)
+#'
+#'  Each option can be overridden with a system environment variable. Options include:
+#'
+#'   - `GEOTARGETS_GDAL_RASTER_CREATION_OPTIONS`
+#'   - `GEOTARGETS_GDAL_RASTER_DRIVER_NAME`
```

Comment:
Would it make sense to have these be a little shorter just for the sake of being concise?
```suggestion
#'   - `GEOTARGETS_GDAL_RASTER_OPTIONS`
#'   - `GEOTARGETS_GDAL_RASTER_DRIVER`
```

Reply:
I think OPTIONS is open ended, and could refer to other parts of this, e.g., "open-options": https://gdal.org/drivers/raster/gtiff.html#open-options

While CREATION_OPTIONS is longer, and I want to reduce typing for the user, I think it might be necessary?

I think RASTER_DRIVER makes sense

Reply:
I think that maybe we can open this discussion as a separate issue? I'm keen to implement this PR

Reply:
Opened #25